app/controllers/api/v1/users_controller.rb

```diff
@@ -33,7 +34,10 @@ def create
       # TODO: Add a before_action callback to update,destory and purge_avatar calling a find_user.
       def update
         user = User.find(params[:id])
+        old_email = user.email
+        # TODO: optimise this.
         if user.update(user_params)
+          user.deactivate! if old_email != user.email
```

Comment:
Let's drop this from this PR since we need to discuss whether we need this or not